### PR TITLE
Update list of PEPs

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -85,6 +85,9 @@ Type-Hint and Stub Integration
 Typing PEPs
 ===========
 
+https://peps.python.org/topic/typing
+
+* `PEP 482 <https://www.python.org/dev/peps/pep-0483/>`_, literature overview on type hints
```
Comment (erictraut) [inline docs/index.rst:90]: The link is incorrect for this one. It's pointing to PEP 483.

File: docs/index.rst [hunk 2]
```diff
@@ -101,8 +104,15 @@ Typing PEPs
 * `PEP 613 <https://www.python.org/dev/peps/pep-0613/>`_, ``TypeAlias``
 * `PEP 646 <https://www.python.org/dev/peps/pep-0646/>`_, variadic generics and ``TypeVarTuple``
 * `PEP 647 <https://www.python.org/dev/peps/pep-0647/>`_, ``TypeGuard``
+* `PEP 649 <https://www.python.org/dev/peps/pep-0698/>`_ (draft), ``from __future__ import co_annotations``
```
Comment (erictraut) [inline docs/index.rst:107]: The link is wrong here too.

**Comment by hauntsaninja** [comment]
Oops, thanks, fixed!

File: docs/index.rst [hunk 3]
```diff
@@ -101,8 +104,15 @@ Typing PEPs
...
-* `PEP 677 <https://www.python.org/dev/peps/pep-0677/>`_ (rejected), callable type syntax
-* `PEP 681 <https://www.python.org/dev/peps/pep-0681/>`_ (draft), ``@dataclass_transform()``
+* `PEP 677 <https://www.python.org/dev/peps/pep-0677/>`_ (rejected), ``(int, str) -> bool`` callable type syntax
+* `PEP 681 <https://www.python.org/dev/peps/pep-0681/>`_ ``@dataclass_transform()``
+* `PEP 688 <https://www.python.org/dev/peps/pep-0688/>`_ (draft), ``Buffer``
+* `PEP 692 <https://www.python.org/dev/peps/pep-0692/>`_ (draft), ``Unpack[TypedDict]`` for ``**kwargs``
+* `PEP 695 <https://www.python.org/dev/peps/pep-0695/>`_ (draft), ``class Class[T]:`` type parameter syntax
+* `PEP 696 <https://www.python.org/dev/peps/pep-0696/>`_ (draft), defaults for type variables
+* `PEP 698 <https://www.python.org/dev/peps/pep-0698/>`_ ``@override``
+* `PEP 702 <https://www.python.org/dev/peps/pep-0702/>`_ ``@deprecated()``
```
Comment (JelleZijlstra) [inline docs/index.rst:118]:
```suggestion
* `PEP 702 <https://www.python.org/dev/peps/pep-0702/>`_ (draft), ``@deprecated()``
```

File: docs/index.rst [hunk 4]
```diff
@@ -101,8 +104,15 @@ Typing PEPs
...
+* `PEP 649 <https://www.python.org/dev/peps/pep-0649/>`_ (draft), ``from __future__ import co_annotations``
 * `PEP 655 <https://www.python.org/dev/peps/pep-0655/>`_, ``Required`` and ``NotRequired``
 * `PEP 673 <https://www.python.org/dev/peps/pep-0673/>`_, ``Self``
 * `PEP 675 <https://www.python.org/dev/peps/pep-0675/>`_, ``LiteralString``
-* `PEP 677 <https://www.python.org/dev/peps/pep-0677/>`_ (rejected), callable type syntax
-* `PEP 681 <https://www.python.org/dev/peps/pep-0681/>`_ (draft), ``@dataclass_transform()``
+* `PEP 677 <https://www.python.org/dev/peps/pep-0677/>`_ (rejected), ``(int, str) -> bool`` callable type syntax
+* `PEP 681 <https://www.python.org/dev/peps/pep-0681/>`_ ``@dataclass_transform()``
+* `PEP 688 <https://www.python.org/dev/peps/pep-0688/>`_ (draft), ``Buffer``
+* `PEP 692 <https://www.python.org/dev/peps/pep-0692/>`_ (draft), ``Unpack[TypedDict]`` for ``**kwargs``
```
Comment (JelleZijlstra) [inline docs/index.rst:114]: This one is accepted but we need to do some implementing I think.

File: docs/index.rst [hunk 5]
```diff
@@ -101,8 +104,15 @@ Typing PEPs
 * `PEP 613 <https://www.python.org/dev/peps/pep-0613/>`_, ``TypeAlias``
 * `PEP 646 <https://www.python.org/dev/peps/pep-0646/>`_, variadic generics and ``TypeVarTuple``
 * `PEP 647 <https://www.python.org/dev/peps/pep-0647/>`_, ``TypeGuard``
+* `PEP 649 <https://www.python.org/dev/peps/pep-0649/>`_ (draft), ``from __future__ import co_annotations``
```
Comment (JelleZijlstra) [inline docs/index.rst:107]: It's not clear we'll actually end up with that future import. Not sure I have a better suggestion though.

File: docs/index.rst (hunk 4 above)
Comment (hauntsaninja) [reply docs/index.rst:114]: Oh great, I thought I'd seen something like that but the PEP still said draft. I'll go ahead and remove the draft here to save some bother.

I think both mypy and pyright fully support it, I can take a look at what typing_extensions / typing need

Comment (hauntsaninja) [inline docs/index.rst:114]:
```suggestion
* `PEP 692 <https://www.python.org/dev/peps/pep-0692/>`_ ``Unpack[TypedDict]`` for ``**kwargs``
```

**Pull request merged**